src/intelli/graph.h

```diff
@@ -375,6 +420,110 @@ class GT_INTELLI_EXPORT Graph : public Node
      */
     bool deleteConnection(ConnectionId connectionId);
 
+    /**
+     * @brief Moves the node (given by NodeId) from this graph to the target
+     * graph. Depending on the given `NodeIdPolicy` the node's id may be
+     * updated. Failing to move a node will not restore the previous state
+     * of the graph. Use a MementoDiff for this purpose. The underlying node
+     * object is guranteed to be kept alive.
+     * Note: All connections to this node are destroyed.
+     * @param nodeId Node to move to other graph
+     * @param policy Whether to update the node's id if necessary (if the
+     * node id should be kept, moving the node may fail)
+     * @return success
+     */
+    bool moveNode(NodeId nodeId,
+                  Graph& targetGraph,
+                  NodeIdPolicy policy = NodeIdPolicy::Update)
+    {
+        Node* node = findNode(nodeId);
+        return node && moveNode(*node, targetGraph, policy);
+    }
+
+    /**
+     * @brief Overload that the node given by reference.
+     * @param nodeId Node to move to other graph
+     * @param policy Whether to update the node's id if necessary (if the
+     * node id should be kept, moving the node may fail)
+     * @return success
+     */
+    bool moveNode(Node& node,
+                  Graph& targetGraph,
+                  NodeIdPolicy policy = NodeIdPolicy::Update);
+
+    /**
+     * @brief Moves the nodes from this graph to the target graph. The
+     * connections inbetween these nodes are moved as well. Depending
+     * on the given `NodeIdPolicy` the node's ids may be updated. Failing to
+     * move a node may will not restore the previous state of the graph. The
+     * underlying node objects are guranteed to be kept alive whereas the
+     * connection objects are deleted. Thus, raw pointers to connection objects
+     * must be updated.
+     * @param nodes Nodes to move
+     * @param targetGraph Target graph to move nodes to
+     * @param policy Whether to generate a new id if necessary
+     * @return success
+     */
+    template <typename NodeList>
+    bool moveNodesAndConnections(NodeList const& nodes,
+                                 Graph& targetGraph,
+                                 NodeIdPolicy policy = NodeIdPolicy::Update)
+    {
+        auto changeThis   = modify();
+        auto changeTarget = targetGraph.modify();
+        Q_UNUSED(changeThis);
```

Comment:
why do you set up variables and do not use them?

Reply:
This is a scoped object, that basically stops execution updates while it is alive. I thus only have to instantiate it and do not have to use it again (It is a `gt::finally` function)

Reply:
Ok. Then please do not use auto here to make it clear to the reader

Reply:
I'll rename the variable and replace auto with its type